questionnaire/dataset/dataset-routes.test.js

```diff
@@ -0,0 +1,148 @@
+            );
+        }
+    }))
+);
+
+// app has an indirect dependency on questionnaire-dal.js, require it after
+// the mock so that it references the mocked version
+const app = require('../../app');
+
+expect.extend(
+    matchersWithOptions({
+        allErrors: true,
+        jsonPointers: true,
+        format: 'full',
+        coerceTypes: false
+    })
+);
+
+describe('/questionnaires/{questionnaireId}/dataset', () => {
+    describe('Given the header "Accept-Version: 1.0.0"', () => {
+        it('should return a corresponding "Content-Version: 1.0.0" header', async () => {
+            const res = await request(app)
+                .get('/api/v1/questionnaires/cb2eadc7-9189-4c4d-9eb3-896b72a1ed16/dataset')
+                .set('Authorization', `Bearer ${tokens['read:questionnaires']}`)
+                .set('Accept-Version', '1.0.0');
+
+            expect(res.get('Content-Version')).toBe('1.0.0');
+        });
+
+        it('should return v1.0.0 of the dataset resource', async () => {
+            const res = await request(app)
+                .get('/api/v1/questionnaires/cb2eadc7-9189-4c4d-9eb3-896b72a1ed16/dataset')
+                .set('Authorization', `Bearer ${tokens['read:questionnaires']}`)
+                .set('Accept-Version', '1.0.0');
+            const v1Schema = await $RefParser.dereference(v1SchemaFilepath);
+
+            expect(res.status).toEqual(200);
+            expect(res.body).toMatchSchema(v1Schema);
+        });
+    });
+
+    describe('Given the header "Accept-Version: 2.0.0"', () => {
+        it('should return a corresponding "Content-Version: 2.0.0" header', async () => {
+            const res = await request(app)
+                .get('/api/v1/questionnaires/cb2eadc7-9189-4c4d-9eb3-896b72a1ed16/dataset')
+                .set('Authorization', `Bearer ${tokens['read:questionnaires']}`)
+                .set('Accept-Version', '2.0.0');
+
+            expect(res.get('Content-Version')).toBe('2.0.0');
+        });
+
+        it('should return v2.0.0 of the dataset resource', async () => {
+            const res = await request(app)
+                .get('/api/v1/questionnaires/cb2eadc7-9189-4c4d-9eb3-896b72a1ed16/dataset')
+                .set('Authorization', `Bearer ${tokens['read:questionnaires']}`)
+                .set('Accept-Version', '2.0.0');
+            const v2Schema = await $RefParser.dereference(v2SchemaFilepath);
+
+            expect(res.status).toEqual(200);
+            expect(res.body).toMatchSchema(v2Schema);
+        });
+    });
+
+    describe('Given no "Accept-Version" header', () => {
+        it('should return the latest version in the "Content-Version" header', async () => {
+            const res = await request(app)
+                .get('/api/v1/questionnaires/cb2eadc7-9189-4c4d-9eb3-896b72a1ed16/dataset')
+                .set('Authorization', `Bearer ${tokens['read:questionnaires']}`);
+
+            expect(res.get('Content-Version')).toBe('2.0.0');
```

Comment:
Not quite following this here: 

if `setDefaultResourceVersion` returns `1.0.0` how is this `2.0.0`

Reply:
The `setDefaultResourceVersion` only toggles on in non-test environments. The tests are run with [this environment](https://github.com/CriminalInjuriesCompensationAuthority/data-capture-service/blob/02125802620b16d8039ea4d2c88ebd99f992ffab/jest.config.js#L14)